oauthproxy.go

```diff
@@ -1185,14 +1189,27 @@ func checkAllowedEmails(req *http.Request, s *sessionsapi.SessionState) bool {
 
 // encodedState builds the OAuth state param out of our nonce and
 // original application redirect
-func encodeState(nonce string, redirect string) string {
-	return fmt.Sprintf("%v:%v", nonce, redirect)
+func encodeState(nonce string, redirect string, shouldEncode bool) string {
+	rawString := fmt.Sprintf("%v:%v", nonce, redirect)
+	if shouldEncode {
+		return base64.RawURLEncoding.EncodeToString([]byte(rawString))
+	} else {
+		return rawString
+	}
 }
 
 // decodeState splits the reflected OAuth state response back into
 // the nonce and original application redirect
-func decodeState(req *http.Request) (string, string, error) {
-	state := strings.SplitN(req.Form.Get("state"), ":", 2)
+func decodeState(req *http.Request, shouldEncode bool) (string, string, error) {
+	rawState := req.Form.Get("state")
+
+	toParse := rawState
```

Comment:
This seems like an unnecessary assignment. Please remove this variable.

Reply:
Sorry, I'm not really a fluent in Go. I was led to believe that this pattern is a standard Go way of doing conditional asignment
``` Go
result := someValue
if something {
    result := someOtherValue
}
```